### PR TITLE
Fixed a bunch of issues with the new external formatting service.

### PR DESCRIPTION
**Pull request by ryan-rushton**
## Overview

This fixes a few issues introduced by the new external formatting service implementation that hooks into the standard intellij formatting commands. I have done the following

1. Put a synchronization lock around the code blocks that call the editor service. Now we are calling it from a new places we need to ensure that it doesn't get called in asynchronously with subsequent calls interrupting the communication process.
2. Made a bunch of the noisy logging debug
3. Perform no-op's when range formatting is requested. Without this IntelliJ reverts back to default formatting and we get differences in the output of whole file vs ranged.
4. Ensured that the file we are formatting is a virtual file. IntelliJ will send text to the format when it doesn't actually exist in a file yet, this happens when refactoring code, automatically generating methods, ect.

### CODE REVIEW

File: src/main/kotlin/com/dprint/services/DprintService.kt [hunk 1]
```diff
@@ -274,28 +284,25 @@ class DprintService(private val project: Project) {
     fun fmt(filePath: String, content: String): DprintResult {
         val result = DprintResult()
 
-        if (!canFormat(filePath)) {
-            result.error = Bundle.message("formatting.cannot.format", filePath)
-            return result
-        }
```
Comment (ryan-rushton) [inline src/main/kotlin/com/dprint/services/DprintService.kt:280]: Decided to pull this out and let consumers call this method themselves

File: src/main/kotlin/com/dprint/services/DprintService.kt [hunk 2]
```diff
@@ -245,23 +247,31 @@ class DprintService(private val project: Project) {
...
+        var status = 0
         LOGGER.info(Bundle.message("formatting.checking.can.format", filePath))
         getEditorService()?.let { editorService ->
             val stdin = editorService.outputStream
             val stdout = editorService.inputStream
 
-            writeInt(stdin, CHECK_COMMAND)
-            writeString(stdout, stdin, filePath)
-            writeSuccess(stdin)
+            synchronized(this) {
```
Comment (dsherret) [inline src/main/kotlin/com/dprint/services/DprintService.kt:256]: Will be so much nicer once I implement https://github.com/dprint/dprint/pull/482

I'm going to start on that one soon.

**Comment by ryan-rushton** [comment]
@dsherret I was using this Friday and it was all working fine for me. I haven't heard back from others so I am happy to push this one out as a release.

**Review by dsherret — APPROVED** [review]
Sounds good!

**Pull request merged**